client/service/src/task_manager/tests.rs

```diff
@@ -315,92 +319,90 @@ fn ensure_task_manager_future_continues_when_childs_not_essential_task_fails() {
+		let (subscriber, spans_found) = setup_subscriber();
+		tracing_log::LogTracer::init().unwrap();
+		let _sub_guard = tracing::subscriber::set_global_default(subscriber);
 
-	let telemetry_span = TelemetrySpan::new();
+		let mut runtime = tokio::runtime::Runtime::new().unwrap();
 
-	let span = tracing::info_span!("test");
-	let _enter = span.enter();
+		let prefix_span = tracing::info_span!("prefix");
+		let _enter_prefix_span = prefix_span.enter();
 
-	let mut runtime = tokio::runtime::Runtime::new().unwrap();
-	let handle = runtime.handle().clone();
-	let task_executor = TaskExecutor::from(move |fut, _| handle.spawn(fut).map(|_| ()));
-	let task_manager = TaskManager::new(task_executor, None, Some(telemetry_span.clone())).unwrap();
+		let telemetry_span = TelemetrySpan::new();
+		let _enter_telemetry_span = telemetry_span.enter();
 
-	let (sender, receiver) = futures::channel::oneshot::channel();
-	let spawn_handle = task_manager.spawn_handle();
+		let handle = runtime.handle().clone();
+		let task_executor = TaskExecutor::from(move |fut, _| handle.spawn(fut).map(|_| ()));
+		let task_manager = new_task_manager(task_executor);
 
-	let span = span.clone();
-	task_manager.spawn_handle().spawn(
-		"test",
-		async move {
-			assert_eq!(span, Span::current());
-			spawn_handle.spawn("test-nested", async move {
-				assert_eq!(span, Span::current());
+		let (sender, receiver) = futures::channel::oneshot::channel();
+
+		task_manager.spawn_handle().spawn(
+			"log-something",
+			async move {
+				log::info!("boo!");
 				sender.send(()).unwrap();
-			}.boxed());
-		}.boxed(),
-	);
+			}
+			.boxed(),
+		);
 
-	// We need to leave exit the span here. If tokio is not running with multithreading, this
-	// would lead to duplicate spans being "active" and forwarding the wrong one.
-	drop(_enter);
-	runtime.block_on(receiver).unwrap();
-	runtime.block_on(task_manager.clean_shutdown());
-	drop(runtime);
-
-	let spans = spans_entered.lock();
-	// We entered the telemetry span and the "test" in the future, the nested future and
-	// the "test" span outside of the future. So, we should have recorded 3 spans.
-	assert_eq!(5, spans.len());
-
-	assert_eq!(spans[0], "test");
-	assert_eq!(spans[1], telemetry_span.span().metadata().unwrap().name());
-	assert_eq!(spans[2], "test");
-	assert_eq!(spans[3], telemetry_span.span().metadata().unwrap().name());
-	assert_eq!(spans[4], "test");
+		runtime.block_on(receiver).unwrap();
+		runtime.block_on(task_manager.clean_shutdown());
+
+		let spans = spans_found.lock().take().unwrap();
+		assert_eq!(2, spans.len());
+
+		assert_eq!(spans[0], prefix_span.id().unwrap());
+		assert_eq!(spans[1], telemetry_span.span().id().unwrap());
+	}
+}
+
+#[test]
+fn telemetry_span_is_forwarded_to_task() {
+	let executable = env::current_exe().unwrap();
+	let output = std::process::Command::new(executable)
+		.env("ENABLE_LOGGING", "1")
+		.args(&["--nocapture", "log_something"])
+		.output()
+		.unwrap();
```

Comment:
This is not testing if anything is printed or whatever. I don't see the reason for this pr.

Reply:
It's checking the exit code 3 lines below. The command fails if the test in the subcommand fails. It works great :+1: Or did I miss something?

Reply:
I mean you log something, but you don't check if that is logged actually.

Reply:
That's because I really badly named the function lol I'm not testing the log, I'm testing that the current span and its parent are what I am expecting. That test itself works just fine but you're right I should still add a test with network communication. Hopefully that could even be used to test that multiple nodes can have different telemetries. That would be perfect

Reply:
> Hopefully that could even be used to test that multiple nodes can have different telemetries.

This isn't in this PR right? It would be very good to have.

Reply:
Yes it will be in the "rework" PR. I think haven't made it yet but the branch is going well